pmultiqc/modules/quantms/quantms.py

```diff
@@ -911,201 +956,15 @@ def dis_decoy(ProteinName):
-                else:
-                    proteins = list(filter(lambda x: not x.endswith(config.kwargs['decoy_affix']), proteins))
-                peptides = set(data[(data['Reference'] == i) & (data['DECOY'] == 'TARGET')]['PeptideSequence'].tolist())
-            else:
-                peptides = set(data[data['Reference'] == i]['PeptideSequence'].tolist())
-
-            self.cal_num_table_data[i]['protein_num'] = len(set(proteins))
-            self.cal_num_table_data[i]['peptide_num'] = len(peptides)
-
-            modified_pep = list(filter(lambda x: re.match(r'.*?\(.*\).*?', x) is not None, peptides))
-            self.cal_num_table_data[i]['modified_peptide_num'] = len(modified_pep)
-
-        #  table of peptide number per protein
-        prot_pep_map = {}
-        num_pep_per_protein = {}
-        for _, row in data.iterrows():
-            ProteinName = row['ProteinName']
-            if len(ProteinName.split(';')) == 1:
-                if config.kwargs['remove_decoy']:
-                    if config.kwargs['affix_type'] == 'prefix':
-                        if ProteinName.startswith(config.kwargs['decoy_affix']):
-                            continue
-                    else:
-                        if ProteinName.endswith(config.kwargs['decoy_affix']):
-                            continue
-                if ProteinName not in prot_pep_map.keys():
-                    prot_pep_map[ProteinName] = [row['PeptideSequence']]
-                elif row['PeptideSequence'] not in prot_pep_map[ProteinName]:
-                    prot_pep_map[ProteinName].append(row['PeptideSequence'])
-            else:
-                for p in ProteinName.split(';'):
-                    if config.kwargs['remove_decoy']:
-                        if config.kwargs['affix_type'] == 'prefix':
-                            if ProteinName.startswith(config.kwargs['decoy_affix']):
-                                continue
-                        else:
-                            if ProteinName.endswith(config.kwargs['decoy_affix']):
-                                continue
-                    if p not in prot_pep_map.keys():
-                        prot_pep_map[p] = [row['PeptideSequence']]
-                    elif row['PeptideSequence'] not in prot_pep_map[p]:
-                        prot_pep_map[p].append(row['PeptideSequence'])
-
-        for key, value in prot_pep_map.items():
-            if str(len(value)) not in num_pep_per_protein.keys():
-                num_pep_per_protein[str(len(value))] = {'Frequency': 1}
-            else:
-                num_pep_per_protein[str(len(value))]['Frequency'] += 1
-
-        keys = sorted(num_pep_per_protein.items(), key=lambda x: int(x[0]))
-        for k in keys:
-            self.num_pep_per_protein[k[0]] = k[1]
-            self.percent_pep_per_protein[k[0]] = {'Percentage': k[1]['Frequency'] / len(prot_pep_map)}
-        self.Total_Protein_Identified = len(prot_pep_map)
-
-    def parse_out_mzTab(self):
```

Comment:
how are you parsing mztab now?

Reply:
Using pyteomics now, all information is obtained from mzTab, no csv file is needed anymore